src/ir_interpreter/interpret_ir.ml

```diff
@@ -401,13 +409,13 @@ and interpret_exp_mut env exp (k : V.value V.cont) =
             k
         | _ -> assert false
         end
-      | CPSAwait _, [v1; v2] ->
+      | CPSAwait (sort, _), [v1; v2] ->
         assert (not env.flavor.has_await && env.flavor.has_async_typ);
         begin match V.as_tup v2 with
          | [vf; vr] ->
            let (_, f) = V.as_func vf in
            let (_, r) = V.as_func vr in
-           await env exp.at (V.as_async v1)
+           await env exp.at (sort <> T.AwaitFut false)(*FIXME?*) (V.as_async v1)
```

Comment:
FIXME?

Reply:
No need, this code is working just fine for ages.